localstack/services/sqs/sqs_listener.py

```diff
@@ -34,14 +34,14 @@ class ProxyListenerSQS(ProxyListener):
 
     def forward_request(self, method, path, data, headers):
 
-        if method == 'POST' and path == '/':
+        if method == 'POST':
             req_data = urlparse.parse_qs(to_str(data))
             if 'QueueName' in req_data:
                 encoded_data = urlencode(req_data, doseq=True)
                 request = Request(data=encoded_data, headers=headers, method=method)
                 return request
             elif req_data.get('Action', [None])[0] == 'SendMessage':
-                queue_url = req_data.get('QueueUrl', [None])[0]
+                queue_url = req_data.get('QueueUrl', path)[0]
```

Comment:
I think this should be wrapped in an array, otherwise we're only selecting the first character of `path`:
```
queue_url = req_data.get('QueueUrl', [path])[0]
```

Reply:
I tried your suggestion and the test still fails, however it only fails if the test is run with USE_SSL=1, It may be because the queue_url is missing the http(s)://localhost:4576 since the path likely does not contain that part.

Reply:
Looks like the `[path]` is '/' if it's https, although I don't know why.

Reply:
Nevermind the path is correct, I was using the AWS client which always hits /, still haven't figured out the issue

Reply:
The issue that caused the failing test was the Lambda when constructing the client a la TestUtils.getClientS3SSL() wasn't using the USE_SSL flag since it was outside the normal flow.

I'll clean up some fixes and check this in tomorrow.